specs/pk_mode.go

```diff
@@ -0,0 +1,50 @@
+package specs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+type PKMode int
+
+const (
+	PKModeCompositeKeys PKMode = iota
+	PKModeCQID
+)
+
+var (
+	pkModeStrings = []string{"composite-keys", "cq-id"}
```

Comment:
```suggestion
	pkModeStrings = []string{"composite-keys", "cq-id"}
```

I'm not sure `composite-keys` is accurate when there's a single PK.
Maybe call these `default` and `cq-id-only`?